x/subscription/spec/03_messages.md

```diff
@@ -0,0 +1,94 @@
+<!--
+order: 3
+-->
+
+# Messages
+
+## MsgCreatePlan
+
+Merchant or payment provider create plans.
+
+```protobuf
+message MsgCreatePlan {
+  string title;
+  string description;
+  Decimal price,  // price in fiat currency show to user
+  string fiat_currency,  // fiat currency name
+  string token_denom,  // denomination of token
+  Decimal conversion_rate,  // conversion rate from fiat currency to token, maintained by owner
+  int32 interval
+}
+```
+
+The message signer is the owner.
+
+## MsgStopPlan
+
+```protobuf
+message MsgStopPlan {
+  int32 planId = 1,
+}
+```
+
+Can only stopped by the owner of the plan, also removes all subscriptions.
+
+## MsgChangeConversionRate
+
+```protobuf
+message MsgChangeConversionRate {
+  int32 planId = 1,
+  Decimal rate = 2,
+}
+```
+
+Can only be changed by plan owner.
+
+## MsgCreateSubscription
+
+```protobuf
+message MsgCreateSubscription {
+  int32 planId = 1,
+}
+```
+
+The message signer subscribe the plan.
+
+## MsgStopSubscription
+
+```protobuf
+message MsgStopSubscription {
+  int32 planId = 1,
+  Address subscriber = 2,
+}
+```
+
+The subscriber unsubscribe the plan, or plan owner cancel a subscription.
+
+## MsgCollectPayments
+
```

Comment:
@christopher-crypto in the real world, there's a "tolerance" -- where e.g. after 3 fails, it'll auto-cancel the subscription

Reply:
but both options exist -- e.g. merchant may not want to auto-cancel

Reply:
Merchant can set `tolerate_failure` to `false`, then it'll automatically cancel failed subscription.
Or set `tolerate_failure` to `true`, then manually handle the retrying.